taf/updater/updater_pipeline.py

```diff
@@ -506,13 +522,28 @@ def load_target_repositories(self):
                 only_load_targets=True,
                 excluded_target_globs=self.excluded_target_globs,
             )
-            self.state.target_repositories = (
+            self.state.users_target_repositories = (
                 repositoriesdb.get_deduplicated_repositories(
                     self.state.users_auth_repo,
                     self.state.auth_commits_since_last_validated[-1::],
                 )
             )
-            return UpdateStatus.SUCCESS
+            if self.only_validate:
+                self.state.temp_target_repositories = (
+                    self.state.users_target_repositories
+                )
+            else:
+                self.state.temp_dir = TemporaryDirectory()
```

Comment:
I don't think we should create a temporary directory at default disk location, right? Because it's often the case that the clone crosses filesystem boundaries. This isn't a problem for the authentication repo, which is often much smaller. But it can be a significant impact to have write operations from D:// to C:// disk. So the clone is expected to be much slower, IIUC. We can solve by creating a temporary directory at the same disk/space that the archive is located at

Reply:
It could help, I pushed these changes.